src/Stack/Build/Execute.hs

```diff
@@ -896,3 +899,26 @@ getSetupHs dir = do
   where
     fp1 = dir </> $(mkRelFile "Setup.hs")
     fp2 = dir </> $(mkRelFile "Setup.lhs")
+
+-- | Update the symlink in @.stack-work\/install\/last@ to point to the
+-- current local install directory.
+addLastInstallSymlink :: M env m => m ()
+addLastInstallSymlink = do
+  target <- installationRootLocal
+  source <- lastInstallSymlink
+  updateSymlink target source
+
+-- | Update the symlink in @.stack-work\/dist\/last@ to point to the
+-- current local install directory.
+addLastDistSymlink :: M env m => m ()
+addLastDistSymlink = do
+  bc <- asks getBuildConfig
+  target <- distDirFromDir (bcRoot bc)
+  source <- lastInstallSymlink
+  updateSymlink target source
+
+updateSymlink :: M env m => Path Abs Dir -> Path Abs File -> m ()
+updateSymlink target source = do
+  createTree target
+  removeFileIfExists source
+  liftIO $ createSymbolicLink (toFilePath target) (toFilePath source)
```

Comment:
As I mentioned in the issue, the implementation of this function on windows just throws an exception.